functions/tokenservice/functions/index.js

```diff
@@ -149,21 +149,17 @@ exports.getOAuthToken = functions.https.onCall(async (data, context) => {
   // Retrieve the token from the database
   const docRef = db.collection('ShortLivedAuthTokens').doc('OauthToken');
 
-  const doc = await docRef.get();
-
   try {
+    const doc = await docRef.get();
     if (doc.exists && isValid(doc.data().expireTime)) {
-      //push notification
       pushNotification(
         data['deviceID'],
         doc.data().accessToken,
         doc.data().expireTime
       );
-      return doc.data();
+      await doc.data();
```

Comment:
This seems confusing - why are we using `doc.data()` presumably _before_ we've `await`ed it? (If we know this case _definitely won't happen_, we should at least leave a comment explaining that.)

cc @santhoshvaddi

Reply:
After push notification we just want to return from the function. The client will receive the token in the form of push notification. I don't think `await` is required here.